### DISCUSSION TITLE
Close gap on VS Code API support

### DISCUSSION THREAD

**Discussion by thegecko** [eclipse-theia/theia#10733] [Improvements]
Most know there are missing features tracked by https://eclipse-theia.github.io/vscode-theia-comparator/status.html.

However, there are also some subtleties in the supported areas which lead to existing extensions not always working properly.

Things I'm aware of (and yes, I know I should raise them as issues!)
- https://github.com/eclipse-theia/theia/issues/10746
- https://github.com/eclipse-theia/theia/issues/10778
- ~~theia.Uri functions fail in frontend plugins (e.g. joinPath)~~ https://github.com/eclipse-theia/theia/pull/10745
- ~~context.extensionUri is incorrect in frontend plugins (its a file path not a url)~~ https://github.com/eclipse-theia/theia/pull/10747
- https://github.com/eclipse-theia/theia/issues/10841 (may be out of scope)

Should we try some top extensions and go on a bug hunt?

**Comment by JonasHelming** [comment]
We could move this to Theia, get it to work again and scan the most popular extension, too
https://github.com/theia-ide/theia-vscodecov
Will not identity bugs, but find missing API.
@planger

**Reply by planger** [reply]
I would also find it very useful to try reviving theia-vscodecov in order to regularly check the most popular extensions. This would give us a prioritization on which parts of missing API should be addressed first. Also it may avoid time-consuming manual checks whether an extension is even supposed to work.
